libsawtooth/src/receipt/store/error.rs

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+use std::error::Error;
+use std::fmt::Debug;
+
+#[derive(Debug)]
+pub enum ReceiptStoreError {
```

Comment:
This might be a good time to copy the InternalError and InvalidStateErrors over from grid and transact here. Would simplify this to two variants.

Reply:
Probably should also move over ResourceTemporarilyUnavailableError as that can be used when adding sqlite/postgress when a pool error is returned.

Reply:
Added a commit to copy those three over and updated the ReceiptStoreError